### ISSUE TITLE
Unparseable C++ cross-reference when referencing a function with :cpp:any:

### ISSUE THREAD

**Issue by vitaut** [sphinx-doc/sphinx#2178]
The following example

``` rst
.. cpp:function:: void f()

:cpp:any:`f()`
```

gives a warning:

```
WARNING: Unparseable C++ cross-reference: u'f()'
Invalid definition: Expected end of definition. [error at 1]
  f()
  -^
```

and the reference is not generated.

Although the warning can be worked around by removing the parentheses, it would be useful if `:cpp:any:` accepted them as `:cpp:func:` does. Otherwise one has to duplicate function name in `:cpp:any:` to render the parentheses:

``` rst
:cpp:any:`f() <f>`
```

Accepting parentheses would also improve consistency with the Python domain where one can do 

``` rst
:py:obj:`f()`
```

**Comment by tk0miya** [comment]
I fixed the problem by another solution at af2ae22.
(to enable add_function_parentheses on :cpp:any: role)

So I close this PR.

**Comment by vitaut** [comment]
Thanks!
